<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
        <h1>Promise</h1>

        <script>
                //    const promiseOne = new Promise(function(resolve,reject){
                //     // DO an async task 
                //     // DB calls, cryptography,network
                //     setTimeout(() => {
                //           console.log('Async task is complete');
                //           resolve()
                //     }, 1000);
                //    })

                //    promiseOne.then(function(){
                //        console.log("Promise consumed");
                //    })


                //      new Promise(function(resolve,reject){
                //         setTimeout(()=>{
                //                   console.log('once again Async');
                //                   resolve()
                //         },1000);
                //      }).then(function(){
                //         console.log("ance again Promised Consumed")
                //      })

                //      const promiseThree = new Promise(function(resolve,reject){
                //                   setTimeout(function(){
                //                     resolve({user_name:"Gill",email:"gill@example.com"})
                //                   },1000)
                //      })
                         
                //      promiseThree.then(function(user){
                //         console.log(user)
                //      })

                     const promiseFour = new Promise(function(resolve,reject){
                        setTimeout(function(){
                            let error = false 
                            if (!error){
                                resolve({user_name:"vashuki",password:"2345"})
                            }else{
                                reject('ERROR:Something went wrong')
                            }
                        },1000)
                     })

                             promiseFour
                             .then((user)=>{
                                     console.log(user)
                                     return user.user_name;
                     })
                     .then((user_name)=>{
                            console.log(user_name);
                     })
                     .catch(function(error ){
                        console.log(error);
                     }).finally(()=>console.log("The promise is either resolved or rejected"))
                              

                     const promiseFive = new Promise(function(resolve,reject){
                            setTimeout(function(){
                                let error = true
                                if (!error){
                                    resolve({user_name:"Harry",password:"456"})                                 
                                }else{
                                    reject('ERROR: Something went wrong')
                                }
                            },1000)
                     });
                                   
                        async function consumePromiseFive() 
                                                     {
                                         try{               
                                      const response = await promiseFive
                                      console.log(response);
                                         }catch(error){
                                            console.log(error);
                                         }                  
                                    }            
                                consumePromiseFive()  
                         
        </script>       
</body>
</html>